src/lib/import-export/import/handlers/backup-handler-zip.ts

```diff
@@ -1,41 +1,92 @@
 import { EventEmitter } from 'events';
-import AdmZip from 'adm-zip';
+import * as fs from 'fs';
+import * as path from 'path';
+import { promisify } from 'util';
+import * as fse from 'fs-extra';
+import * as yauzl from 'yauzl';
 import { ImportEvents } from '../events';
 import { BackupArchiveInfo, BackupExtractProgressEventData } from '../types';
 import { BackupHandler, isFileAllowed } from './backup-handler-factory';
 
+const openZip = promisify< string, yauzl.Options, yauzl.ZipFile >( yauzl.open );
+
 export class BackupHandlerZip extends EventEmitter implements BackupHandler {
 	async listFiles( backup: BackupArchiveInfo ): Promise< string[] > {
-		const zip = new AdmZip( backup.path );
-		return zip
-			.getEntries()
-			.map( ( entry ) => {
-				if ( entry.entryName.startsWith( '/' ) ) {
-					return entry.entryName.slice( 1 );
+		const zipFile = await openZip( backup.path, { lazyEntries: true } );
+		const fileNames: string[] = [];
+
+		return new Promise( ( resolve, reject ) => {
+			zipFile.on( 'entry', ( entry ) => {
+				if ( isFileAllowed( entry.fileName ) ) {
+					fileNames.push( entry.fileName );
 				}
-				return entry.entryName;
-			} )
-			.filter( isFileAllowed );
+				zipFile.readEntry();
+			} );
+
+			zipFile.on( 'end', () => {
+				resolve( fileNames );
+			} );
+
+			zipFile.on( 'error', reject );
+			zipFile.readEntry();
+		} );
 	}
 
 	async extractFiles( file: BackupArchiveInfo, extractionDirectory: string ): Promise< void > {
+		const zipFile = await openZip( file.path, { lazyEntries: true } );
```

Comment:
```suggestion
		const zipFile = await openZip( file.path, { lazyEntries: true } );
		const openReadStream = promisify( zipFile.openReadStream.bind( zipFile ) );
```
And something like this for promisifying `openReadStream`